### ISSUE TITLE
Sampler function seems "indecisive" in lstm_text_generation example

### ISSUE THREAD

**Issue by marczellm** [fchollet/keras#6477]
This function: 
```
def sample(preds, temperature=1.0):
    # helper function to sample an index from a probability array
    preds = np.asarray(preds).astype('float64')
    preds = np.log(preds) / temperature
    exp_preds = np.exp(preds)
    preds = exp_preds / np.sum(exp_preds)
    probas = np.random.multinomial(1, preds, 1)
return np.argmax(probas)
```

transforms the distribution like this:

    f(p, T) = exp(ln(p)/T) = exp(ln(p)) ^ (1/T) = p ^ (1/T)

so it seems like an oversight to do the logarithm and then exponentiate back again.

Now `p^(1/T)` and `exp(p/T)` are both valid transforms for this task, see [Softmax sampling at Wikipedia](https://en.wikipedia.org/wiki/Softmax_function#Reinforcement_learning) and [Stanford](https://nlp.stanford.edu/blog/maximum-likelihood-decoding-with-rnns-the-good-the-bad-and-the-ugly/), but which one was intended here?

**Comment by aaleotti-unimore** [comment]
I would like clarification on this one as well, please
